bookmarks/vanilla/src/modules/Bookmark.ts

```diff
@@ -156,90 +164,81 @@ export class Bookmark {
      * @param title string
      * @param link string
      * @param session Session
-     * @returns IBookmark[]
+     * @returns IBookmark
      */
-    public static async update(url: string, title: string, link: string, session: Session) {
+    public static async update(url: string, payload: IUpdateBookmark, session: Session) {
         const indexUrl = await this.getIndexUrl(session);
         const ds = await getSolidDataset(indexUrl, { fetch: session.fetch });
-        const thing = getThing(ds, url)
+        let thing = getThing(ds, url)
 
         if (thing) {
-            let updatedBookmarkThing = buildThing(thing)
-                .setStringNoLocale(DCTERMS.title, title)
-                .setStringNoLocale(BOOKMARK.recalls, link)
-                .setUrl(RDF.type, BOOKMARK.Bookmark)
-                .build();
-
-            const updatedBookmarkList = setThing(ds, updatedBookmarkThing);
-            const updatedDataset = await saveSolidDatasetAt(indexUrl, updatedBookmarkList, { fetch: session.fetch });
-            const things = getThingAll(updatedDataset)
-
-            return things.map(thing => {
-                return {
-                    url: thing.url,
-                    title: getLiteral(thing, DCTERMS.title)?.value,
-                    link: getLiteral(thing, BOOKMARK.recalls)?.value
-                }
-            }) as IBookmark[]
+            const { title, link, creator, topic } = payload
+
+            thing = setStringNoLocale(thing, DCTERMS.title, title)
```

Comment:
What if the caller only wants to change one of the fields? I think we have to check and only try to update the values for the fields that are assigned.

Reply:
this is an API change, and it's not what this PR is aimed to do,
we have the interface for payloads, so with the current implementation you have to pass the whole object as a payload
so basically it is a `PUT` and not a `PATCH` 

but maybe we can open an issue to talk about it there

Reply:
basically we expect the required fields to exist in payload, because its a PUT, not a partial update

Reply:
You're right. Makes sense to me.